test/test_nestedtensor.py

```diff
@@ -446,6 +446,56 @@ def test_clone(self, device, dtype):
         with self.assertRaisesRegex(RuntimeError, msg):
             nt1.clone(memory_format=torch.channels_last)
 
+class TestNestedTensorAutograd(TestCase):
+    def _create_nested_tensor_from_mask(self, requires_grad=False):
+        return torch.nested_tensor([torch.randn(1, 2, requires_grad=requires_grad),
+                                    torch.randn(7, 8, requires_grad=requires_grad)])
+
+    def _create_nested_tensor_from_list(self, requires_grad=False):
+        data = torch.randn(2, 3, 4, requires_grad=requires_grad)
+        mask = torch.ones_like(data[:, :, 0]).bool()
+        return torch._nested_tensor_from_mask(data, mask)
+
+    def test_requires_grad(self):
+        nt = self._create_nested_tensor_from_list()
+        nt.requires_grad_()
+        assert nt.requires_grad
+
+    def test_requires_grad_from_mask(self):
+        nt = self._create_nested_tensor_from_mask()
+        nt.requires_grad_()
+        assert nt.requires_grad
+
+    def test_requires_grad_not_pass_through(self):
+        # I don't know if this is behavior is intended
+        nt = self._create_nested_tensor_from_list(True)
+        assert not nt.requires_grad
+
+    def test_requires_grad_not_pass_through_from_mask(self):
+        # I don't know if this is behavior is intended
```

Comment:
This is not the desired behavior but I marked this function as non_differentiable for this PR. The next PR, which will be created from taking the important bits of the [Draft] PR, will implement the derivative functions for the factory functions that create nested tensors from "regular" tensors. 

In that case the intermediate nt will also require_grad

Reply:
Why is it the desired behavior for gradients to flow through the boundary between nested tensors and the inner tensors?

Reply:
I don't think this is strictly necessary for the standard training loop. Grab padded batch and mask (neither require grad)-> convert to nested -> run inference -> update model params. Since the data will likely not require grad. However there is likely some use case where a user would want the grads to flow through this boundary. I also think this would be a helpful way of doing gradcheck